File: planning-idsdp-rms-frontend/src/app/main/modules/training-institute/services/participant.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from "@angular/common/http";
import { PaginationResponse } from "../models/pagination-response.model";
import { environment } from "../../../../../environments/environment";
import { ParticipantModel } from '../models/participant.model';
import { Observable } from "rxjs";
import { map } from 'rxjs/operators';
import { ApiResponseModel } from "../models/api-response.model";

@Injectable({
    providedIn: 'root'
})
export class ParticipantService {

    baseTiEndPoint = environment.ibcs.tiBackend;

    constructor(private http: HttpClient) {
    }

    getParticipantList(size: number, page: number) {
        return this.http.get<PaginationResponse<ParticipantModel[]>>(
            this.baseTiEndPoint + "participants?pageNo=" + page + "&pageSize=" + size);
    }
    getAllProposalList(uuid): Observable<any> {
        return this.http.get(this.baseTiEndPoint + "participants/participants-view/" + uuid);
    }

    createParticipant(newParticipantModel: ParticipantModel) {
        return this.http.post<ApiResponseModel>(this.baseTiEndPoint + "participants", newParticipantModel);
    }

    deleteParticipant(id) {
        return this.http.delete(this.baseTiEndPoint + "participants/" + id);
    }

    getParticipantById(courseId: number) {
        return this.http.get<any>(
            this.baseTiEndPoint + "participants/" + courseId);
    }

    editParticipant(newParticipantModel: ParticipantModel, participantId: number) {
        return this.http.put(this.baseTiEndPoint + "participants/" + participantId, newParticipantModel);
    }

    public getParticipantViewById(id: any): Observable<any> {
        return this.http.get(this.baseTiEndPoint + "participants/" + "participants-view/" + id).pipe(map(res => res));
    }

    getParticipantsByCourseId(courseId: number) {
        return this.http.get<ParticipantModel[]>(this.baseTiEndPoint + "participants/courses/" + courseId);
    }

    getParticipantsByProposalId(proposalId: number) {
        return this.http.get<ParticipantModel[]>(this.baseTiEndPoint + "participants/find-by-proposalId/" + proposalId);
    }

    changeCompleteStatus(id, checked: boolean) {
        return this.http.put<any>(this.baseTiEndPoint + 'participants/change-complete-status/'
            + id + "?completeStatus=" + checked, null);
    }
}

File: planning-idsdp-rms-ti-backend/src/main/java/com/ibcs/idsdp/trainninginstitute/services/ParticipantService.java
package com.ibcs.idsdp.trainninginstitute.services;

import com.ibcs.idsdp.trainninginstitute.model.domain.ParticipantModel;
import com.ibcs.idsdp.trainninginstitute.web.dto.request.ParticipantRequest;
import com.ibcs.idsdp.trainninginstitute.web.dto.response.ApiMessageResponse;
import com.ibcs.idsdp.trainninginstitute.web.dto.response.ApiResponse;
import com.ibcs.idsdp.trainninginstitute.web.dto.response.PaginationResponse;
import com.ibcs.idsdp.trainninginstitute.web.dto.response.ParticipantView;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PathVariable;

import java.util.List;

public interface ParticipantService {
	
    ResponseEntity<ApiResponse<Long>> createParticipant(ParticipantRequest participantRequest);

    ResponseEntity<PaginationResponse<List<ParticipantModel>>> getParticipantList(int pageNo, int pageSize, String name);

    ResponseEntity<ApiMessageResponse> editParticipant(Long participantId, ParticipantRequest participantRequest);

    ResponseEntity<ApiMessageResponse> deleteParticipant(Long participantId);

    ResponseEntity<ParticipantModel> getParticipant(Long participantId);

    ParticipantView getParticipantView(Long id);

    ResponseEntity<List<ParticipantModel>> getParticipantsByCourseId(Long courseId);

    ResponseEntity<ApiMessageResponse> changeCompleteStatus(Long participantId, boolean completeStatus);
    
    ResponseEntity<List<ParticipantModel>> getParticipantsByProposalId(Long proposalId);
}
